tools/src/main/java/org/apache/rocketmq/tools/command/controller/CleanControllerBrokerDataSubCommand.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.rocketmq.tools.command.controller;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.apache.rocketmq.remoting.RPCHook;
+import org.apache.rocketmq.tools.admin.DefaultMQAdminExt;
+import org.apache.rocketmq.tools.command.SubCommand;
+import org.apache.rocketmq.tools.command.SubCommandException;
+
+public class CleanControllerBrokerDataSubCommand implements SubCommand {
+
+    @Override
+    public String commandName() {
+        return "cleanBrokerData";
+    }
+
+    @Override
+    public String commandDesc() {
+        return "Clean data of broker which shut down";
+    }
+
+    @Override
+    public Options buildCommandlineOptions(Options options) {
+
+        Option opt = new Option("a", "controllerAddress", true, "The address of controller");
+        opt.setRequired(true);
+        options.addOption(opt);
+
+        opt = new Option("b", "brokerAddress", true, "The address of the broker which requires to clean metadata. eg: 192.168.0.1:30911;192.168.0.2:30911");
+        opt.setRequired(false);
+        options.addOption(opt);
+
+        opt = new Option("n", "brokerName", true, "The broker name of the replicas that require to be manipulated");
+        opt.setRequired(true);
+        options.addOption(opt);
+
+        opt = new Option("c", "clusterName", true, "the clusterName of broker");
```

Comment:
Hi, @mxsm Could you explain the meaning of clusterName here？

Reply:
> Hi, @mxsm Could you explain the meaning of clusterName here？

In Controller server ReplicasInfoManager#cleanBrokerData I need it and brokerAddrs to test broker is active.
```java
# ReplicasInfoManager#cleanBrokerData
    public ControllerResult<Void> cleanBrokerData(final CleanControllerBrokerDataRequestHeader requestHeader,
        final BiPredicate<String, String> brokerAlivePredicate) {
        final ControllerResult<Void> result = new ControllerResult<>();

        final String clusterName = requestHeader.getClusterName();
        final String brokerName = requestHeader.getBrokerName();
        final String brokerAddrs = requestHeader.getBrokerAddress();

        //if SyncStateInfo.masterAddress is not empty, at least one broker with the same BrokerName is alive
        SyncStateInfo syncStateInfo = this.syncStateSetInfoTable.get(brokerName);
        if (StringUtils.isBlank(brokerAddrs) && null != syncStateInfo && StringUtils.isNotEmpty(syncStateInfo.getMasterAddress())) {
            String remark = String.format("Broker %s is still alive, clean up failure", requestHeader.getBrokerName());
            result.setCodeAndRemark(ResponseCode.CONTROLLER_INVALID_REQUEST, remark);
            return result;
        }
        Set<String> brokerAddressSet = null;
        if (StringUtils.isNotBlank(brokerAddrs)) {
            brokerAddressSet = Stream.of(brokerAddrs.split(";")).collect(Collectors.toSet());
            for (String brokerAddr : brokerAddressSet) {
                if (brokerAlivePredicate.test(clusterName, brokerAddr)) {
                    String remark = String.format("Broker [%s,  %s] is still alive, clean up failure", requestHeader.getBrokerName(), brokerAddr);
                    result.setCodeAndRemark(ResponseCode.CONTROLLER_INVALID_REQUEST, remark);
                    return result;
                }
            }
        }
        if (isContainsBroker(brokerName)) {
            final CleanBrokerDataEvent event = new CleanBrokerDataEvent(brokerName, brokerAddressSet);
            result.addEvent(event);
            return result;
        }
        result.setCodeAndRemark(ResponseCode.CONTROLLER_INVALID_REQUEST, String.format("Broker %s is not existed", brokerName));
        return result;
    }
```

Reply:
OK, I got it. But we need to discuss whether we can only clean up non-living brokers.

Reply:
> OK, I got it. But we need to discuss whether we can only clean up non-living brokers.

Can we give the user an option to decide whether only clean up non-living brokers.

Reply:
> > OK, I got it. But we need to discuss whether we can only clean up non-living brokers.
> 
> @RongtongJin Can we give the user an option to decide whether only clean up non-living brokers.

Good suggestion

Reply:
> > > OK, I got it. But we need to discuss whether we can only clean up non-living brokers.
> > 
> > 
> > @RongtongJin Can we give the user an option to decide whether only clean up non-living brokers.
> 
> Good suggestion

I will follow this current idea to implement @RongtongJin